src/Pages/Detail/SupportCard.js

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { FaCheck } from 'react-icons/fa';
+import { FaTimesCircle } from 'react-icons/fa';
+
+const SupportCard = () => {
+  const [card, setCard] = useState([
+    // 레이아웃 점검용 목데이터
+    {
+      id: 1,
+      people: 1123,
+      money: 1,
+      description: '강아지의 간식을 업그레이드 합니다.',
+    },
+    {
+      id: 2,
+      people: 2313,
+      money: 2,
+      description: '강아지의 옷을 업그레이드 합니다.',
+    },
+    {
+      id: 3,
+      people: 948,
+      money: 3,
+      description: '강아지의 집을 업그레이드 합니다.',
+    },
+  ]);
+
+  const [idx, setIdx] = useState(0);
+
+  const openAdditionalSupport = id => {
+    setIdx(id);
+    setMoney(0);
+  };
+
+  const [money, setMoney] = useState(0);
+
+  const selectPlusSupport = e => {
+    setMoney(money + Number(e.target.value));
+  };
+
+  const changeSupportValue = e => {
+    setMoney(Number(e.target.value));
+  };
```

Comment:
만약 5천원, 1만원, 5만원, 10만원 버튼의 타입을 `type="button"`으로 따로 설정해준다면
**가격을 변동시키는 함수**는 하나로 만들 수 있겠네요!
```suggestion
  const changeSupportValue = e => {
    setMoney((e.target.type === 'button' ? money : 0) + Number(e.target.value));
  };
```

Reply:
와! 감사합니다!